### PR TITLE
Add common super class called FileAccessRequest for FileDownloadRequest and a private request used in preview sdk

### PR DESCRIPTION
**Pull request by DJBen**

### CODE REVIEW

**Comment by boxcla** [comment]
Hi @DJBen, thanks for the pull request. Before we can merge it, we need you to sign our Contributor License Agreement. You can do so electronically here: http://opensource.box.com/cla

Once you have signed, just add a comment to this pull request saying, "CLA signed". Thanks!

**Comment by DJBen** [comment]
CLA signed

**Comment by boxcla** [comment]
Verified that @DJBen has just signed the CLA. Thanks, and we look forward to your contribution.

**Comment by RicoYao** [comment]
@boxtcarpel @scottliubox

File: BoxContentSDK/BoxContentSDK/Requests/BOXFileDownloadRequest.m [hunk 1]
```diff
@@ -8,71 +8,20 @@
...
-}
-
 
-- (BOXAPIOperation *)createOperation
-{
-    NSURL *URL = [self URLWithResource:BOXAPIResourceFiles
-                                    ID:self.fileID
-                           subresource:BOXAPISubresourceContent
-                                 subID:nil];
+@interface BOXFileAccessRequest ()
```
Comment (boxtcarpel) [inline BoxContentSDK/BoxContentSDK/Requests/BOXFileDownloadRequest.m]: can be for BOXDownloadRequest, right?

**Comment by DJBen** [comment]
Don't think this change is required. I can just subclass the FileDownloadRequest.

**Pull request closed without merging**